java/arcs/core/crdt/CrdtEntity.kt

```diff
@@ -80,6 +80,35 @@ class CrdtEntity(
                 allOps = false
             }
         }
+
+        if (_data.creationTimestamp != other.creationTimestamp) {
+            allOps = false
+            if (_data.creationTimestamp == RawEntity.UNINITIALIZED_TIMESTAMP) {
+                _data.creationTimestamp = other.creationTimestamp
+            } else if (other.creationTimestamp != RawEntity.UNINITIALIZED_TIMESTAMP) {
+                // Two different values, we keep the earlier creationTimestamp.
+                _data.creationTimestamp = minOf(_data.creationTimestamp, other.creationTimestamp)
+            }
+        }
+        if (_data.expirationTimestamp != other.expirationTimestamp) {
+            allOps = false
+            if (_data.expirationTimestamp == RawEntity.UNINITIALIZED_TIMESTAMP) {
+                _data.expirationTimestamp = other.expirationTimestamp
+            } else if (other.expirationTimestamp != RawEntity.UNINITIALIZED_TIMESTAMP) {
+                // Two different values, we don't how to merge this.
```

Comment:
i think we should use the expiration timestamp of the same data we used creation timestamp of.
and in general, data can't have expiration if it doesn't have creation (i don't think we ever check for it, but it shouldn't be possible).

Reply:
after our discussion, I think throwing here is ok, and we should also throw for creationTimestamps, and these timestamps should be created with the id and then never change. I added a TODO for that as it doesn't work at the moment.

Reply:
agreed, these should all be tightly tied together.

Reply:
I will work on the TODO asap.